### DISCUSSION TITLE
Puppet 7 support for archived mcollective-puppet-agent

### DISCUSSION THREAD

**Discussion by smortex** [choria-io/general#3] [Q&A] [answered]
After updating my infrastructure to Puppet 7.0.0, the puppet agent is non-functional:

```
Choria MCollective Compatibility Layer failed to invoke the action: Cannot manage Puppet version 7
```

Adding version `"7"` to the list here seems to do the trick:
https://github.com/choria-legacy/mcollective-puppet-agent/blob/c4c637165a08d02b0efca4f3654d482ba4463f9d/util/puppet_agent_mgr.rb#L54

But the repo is archived.  How to proceed to fix this?

**Comment by ripienaar** [comment, marked as answer]
I read on slack that some locations to lock files, maybe the lock reason etc has changed we will need to vet all of those and probably add support to derive those automatically if p7 is detected. 

The plug-ins are now in choria-plugins org - legacy is what Puppet Inc donated to me unmodified

**Reply by smortex** [reply]
Ah yes, `choria-plugins`!  I was stuck on `choria-legacy` and `choria-io` ETOOMANYORGANIZATIONS!

**Comment by ripienaar** [comment]
Ah actually on close inspection:

https://github.com/choria-plugins/puppet-agent/blob/master/util/puppet_agent_mgr/mgr_v3.rb

is doing the right thing by loading up puppet settings and asking puppet for these values (also why its so damn slow)

So this is fine, update there to support 7 and it should be golden
